datasette/cli.py

```diff
@@ -52,22 +52,34 @@ def publish(publisher, files, name, metadata, extra_options, force, **extra_meta
 
     Example usage: datasette publish now my-database.db
     """
-    if not shutil.which('now'):
-        click.secho(
-            ' The publish command requires "now" to be installed and configured ',
-            bg='red',
-            fg='white',
-            bold=True,
-            err=True,
-        )
-        click.echo('Follow the instructions at https://zeit.co/now#whats-now', err=True)
-        sys.exit(1)
-
-    with temporary_docker_directory(files, name, metadata, extra_options, extra_metadata):
-        if force:
-            call(['now', '--force'])
-        else:
-            call('now')
+    if publisher == 'now':
+        if not shutil.which('now'):
+            click.secho(
+                ' The publish command requires "now" to be installed and configured ',
+                bg='red',
+                fg='white',
+                bold=True,
+                err=True,
+            )
+            click.echo('Follow the instructions at https://zeit.co/now#whats-now', err=True)
+            sys.exit(1)
+
+        with temporary_docker_directory(files, name, metadata, extra_options, extra_metadata):
+            if force:
+                call(['now', '--force'])
+            else:
+                call('now')
+
+    elif publisher == 'heroku':
+        # FIXME: need to verify we have heroku, heroku-builds, and are logged in (ugh)
```

Comment:
Provided the heroku commands output something sensible and quits if you aren't logged in I think that would be fine.

Reply:
I'll have to test that and see.